frontends/p4/toP4/toP4.cpp

```diff
@@ -1051,25 +1051,35 @@ bool ToP4::preorder(const IR::IfStatement* s) {
     visit(s->condition);
     builder.append(") ");
     if (!s->ifTrue->is<IR::BlockStatement>()) {
+        builder.append("{");
         builder.increaseIndent();
         builder.newline();
         builder.emitIndent();
     }
     visit(s->ifTrue);
-    if (!s->ifTrue->is<IR::BlockStatement>())
+    if (!s->ifTrue->is<IR::BlockStatement>()) {
+        builder.newline();
         builder.decreaseIndent();
+        builder.emitIndent();
+        builder.append("}");
+    }
     if (s->ifFalse != nullptr) {
         builder.newline();
         builder.emitIndent();
         builder.append("else ");
         if (!s->ifFalse->is<IR::BlockStatement>()) {
+            builder.append("{");
```

Comment:
Might be even cleaner to make the if (on lines 1070 and 1077) `if (!s->ifFalse->is<IR::BlockStatement>() && !s->ifFalse->is<IR::IfStatement>())`.  That way chained if ... else if ... else if ... else if ... else ... will line up nicely rather than with increasing indent...